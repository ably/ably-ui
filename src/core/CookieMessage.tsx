import React, { useRef, useEffect, useState } from "react";
import Cookie from "js-cookie";

import _absUrl from "./url-base";

const COOKIE_EXPIRY = 365;

type CookieMessageProps = {
  cookieId: string;
  urlBase: string;
};

const CookieMessage = ({ cookieId, urlBase }: CookieMessageProps) => {
  const ref = useRef<HTMLDivElement>(null);
  const [hideCookieMessage, setHideCookieMessage] = useState(true);

  useEffect(() => {
    const isCookieSet = Cookie.get(cookieId) ? true : false;
    setHideCookieMessage(isCookieSet);
  }, []);

  const handleClose = () => {
    Cookie.set(cookieId, "1", { expires: COOKIE_EXPIRY });

    ref.current?.classList.add("bottom-px", "opacity-0");
    setTimeout(() => setHideCookieMessage(true), 500);
  };

  const absUrl = (path: string) => _absUrl(path, urlBase);

  // Presume the message is hidden by default
  if (hideCookieMessage) return null;

  return (
    <div className="ui-cookie-message" ref={ref}>
      <p className="ui-text-p2 pr-8">
        <a href={absUrl("/privacy")} className="underline">
          How we use cookies
        </a>{" "}
        to improve your experience.
      </p>
      <button
        className="ui-btn-secondary mt-3 sm:mt-0 whitespace-nowrap"
        onClick={handleClose}
      >
        Accept and close
      </button>
    </div>
  );
};

export default CookieMessage;
