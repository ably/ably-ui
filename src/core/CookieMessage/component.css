.ui-cookie-message {
  @apply rounded-lg bg-white font-sans;
  @apply justify-between items-center;
  @apply opacity-100 z-50 bottom-0 fixed sm:flex;
  @apply p-16 mb-16 ml-16;
  max-width: 70vw;
  box-shadow: 0px 24px 32px 0px #0000000d;
  border: 1px solid var(--color-mid-grey);
  border-left: 0.5rem solid var(--color-electric-cyan);
  transition: bottom 250ms ease-out, opacity 150ms ease-out;
}
