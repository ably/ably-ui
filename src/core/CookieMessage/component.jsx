import React, { useRef, useEffect, useState } from "react";
import T from "prop-types";
import Cookie from "js-cookie";

import _absUrl from "../url-base";

const COOKIE_EXPIRY = 365;

export default function CookieMessage({ cookieId, urlBase }) {
  const ref = useRef(null);
  const [hideCookieMessage, setHideCookieMessage] = useState(true);

  useEffect(() => {
    const isCookieSet = Cookie.get(cookieId) ? true : false;
    setHideCookieMessage(isCookieSet);
  }, []);

  const handleClose = () => {
    Cookie.set(cookieId, "1", { expires: COOKIE_EXPIRY });

    ref.current.classList.add("bottom-1", "opacity-0");
    setTimeout(() => setHideCookieMessage(true), 500);
  };

  const absUrl = (path) => _absUrl(path, urlBase);

  // Presume the message is hidden by default
  if (hideCookieMessage) return null;

  return (
    <div className="ui-cookie-message" ref={ref}>
      <p className="ui-text-p2 pr-32">
        <a href={absUrl("/privacy")} className="underline">
          How we use cookies
        </a>{" "}
        to improve your experience.
      </p>
      <button className="ui-btn-secondary mt-12 sm:mt-0 whitespace-nowrap" onClick={handleClose}>
        Accept and close
      </button>
    </div>
  );
}

CookieMessage.propTypes = {
  cookieId: T.string,
  urlBase: T.string,
};
