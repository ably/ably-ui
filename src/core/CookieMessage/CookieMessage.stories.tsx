import CookieMessage from "./component.tsx";

export default {
  title: "Components/Cookie Message",
  component: CookieMessage,
  args: {
    cookieId: "cookie-namespace",
    noticeId: "cookie-message",
  },
};

export const Default = {};
