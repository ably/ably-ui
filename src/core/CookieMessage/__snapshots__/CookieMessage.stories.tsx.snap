// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JS Components/Cookie Message Default smoke-test 1`] = `
<div class="ui-cookie-message">
  <p class="ui-text-p2 pr-32">
    <a href="/privacy"
       class="underline"
    >
      How we use cookies
    </a>
    to improve your experience.
  </p>
  <button class="ui-btn-secondary mt-12 sm:mt-0 whitespace-nowrap">
    Accept and close
  </button>
</div>
`;
